torchvision/datasets/utils.py

```diff
@@ -48,6 +48,19 @@ def _urlretrieve(url: str, filename: str, chunk_size: int = 1024 * 32) -> None:
         _save_response_content(iter(lambda: response.read(chunk_size), b""), filename, length=response.length)
 
 
+def gen_bar_updater() -> Callable[[int, int, int], None]:
+    warnings.warn("The function `gen_bar_update` is deprecated since 0.13 and will be removed in 0.15.", FutureWarning)
```

Comment:
Maybe we should use `FutureWarning` everywhere, but for now we just use regular UserWarning

```suggestion
    warnings.warn("The function `gen_bar_update` is deprecated since 0.13 and will be removed in 0.15.")
```